transport/tchannel/handler.go

```diff
@@ -309,6 +310,12 @@ func (hw *handlerWriter) Close() error {
 	return retErr
 }
 
+func (hw *handlerWriter) ReleaseBuffer() {
+	if hw.buffer != nil {
+		bufferpool.Put(hw.buffer)
```

Comment:
should we set `hw.buffer = nil` after we put it back to ensure we can't put it back in twice?